### PR TITLE
Porting memo spec wasm termination from Osmosis for XCVM ICS-20 memo as program carrier

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
- [x] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [x] I marked PR by `misc` label if it should not be in release notes
- [ ] I have linked Zenhub/Github or any other reference item if one exists
- [x] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [ ] I waited and did best effort for `pr-workflow-check / draft-release-check` to finish with success(green check mark) with my changes
- [x] I have added at least one reviewer in reviewers list
- [ ] I tagged(@) or used other form of notification of one person who I think can handle best review of this PR
- [x] I have proved that PR has no general regressions of relevant features and processes required to release into production

### CODE REVIEW

File: code/parachain/frame/composable-traits/src/ibc.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+#[cfg_attr(feature = "std", derive(JsonSchema))]
+pub enum SudoMsg {
+    #[serde(rename = "ibc_lifecycle_complete")]
+    IBCLifecycleComplete(IBCLifecycleComplete),
+}
+
+pub const SENDER_PREFIX : &str = "ibc-wasm-hook-intermediary";
+
+/// from Go code to make compliant wasm hook
+pub fn derive_intermediate_sender(channel: String, original_sender : String, bech32_prefix : String) -> Result<String, ()> {
```
Comment (mina86) [inline code/parachain/frame/composable-traits/src/ibc.rs:54]: It’s more idiomatic to take the arguments by reference:

```suggestion
pub fn derive_intermediate_sender(channel: &str, original_sender: &str, bech32_prefix: &str) -> Result<String, ()> {
```

This way caller is not forced to make an owned object which this
function discards anyway.

File: code/parachain/frame/composable-traits/src/ibc.rs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+impl MemoData {
+	pub fn forward(forward: Forward) -> Self {
+		Self::Forward(forward)
+	}
+
+	pub fn wasm(wasm: Wasm) -> Self {
+		Self::Wasm(wasm)
+	}
+}
```
Comment (mina86) [inline code/parachain/frame/composable-traits/src/ibc.rs:145]: I probably wouldn’t bother with those methods.  `MemoData::Forward(...)` is just as good as `MemoData::forward(...)`.

**Review by mina86 — APPROVED** [review]

File: code/parachain/frame/composable-traits/src/ibc.rs (hunk 1 above)
Comment (dzmitry-lahoda) [reply code/parachain/frame/composable-traits/src/ibc.rs:54]: yeah, sorry. just direct cp paste from go. did not noticed

File: code/parachain/frame/composable-traits/src/ibc.rs (hunk 2 above)
Comment (dzmitry-lahoda) [reply code/parachain/frame/composable-traits/src/ibc.rs:145]: will remove in next pr if needed

Comment (dzmitry-lahoda) [reply code/parachain/frame/composable-traits/src/ibc.rs:145]: or split per parameter

**Pull request merged**